log_entry.go

```diff
@@ -115,8 +115,27 @@ func (c *Channel) UnmarshalJSON(b []byte) error {
 	ct, ok := raw["type"]
 	if ok {
 		c.Type = ct.(string)
+		delete(raw, "type")
 		c.Raw = raw
 	}
 
 	return nil
 }
+
+// MarshalJSON Expands the LogEntry.Channel object to correctly marshal it back
+func (c *Channel) MarshalJSON() ([]byte, error) {
+	raw := map[string]interface{}{}
+	if c != nil && c.Type != "" {
+		raw["type"] = c.Type
+		for k, v := range c.Raw {
+			raw[k] = v
+		}
+	}
+
+	b, err := json.Marshal(raw)
+	if err != nil {
+		return nil, err
```

Comment:
I suspect `json.Marshal()` effectively does the same thing. How would you feel about simplifying this error handling block all down to `return json.Marshal(raw)`?

Reply:
Nice catch. I'll fix it